### PR TITLE
Tags and Priority

### PR DESCRIPTION
**Pull request by horsemaker**
- Added Priority Field Component
- Added Priority Field in Text Editor
- Added Priority Field in Note Card
- Refactored Editor Reducer for Priority
- Added Tag Screen
- Added Dynamic Tags in Sidebars

recording

### CODE REVIEW

**Review by MiheerTamkhane — APPROVED** [review]
Code is good and very well structured alos the custome hooks part is great apporach.
But on the deploy I got some bugs 
1. When you add empty label / tag and try to delete it. rather to delete its generating new tags.
2. and alos the sidebar label route takes user to home page.

File: src/frontend/components/NoteCard/NoteCard.css [hunk 1]
```diff
@@ -11,11 +11,22 @@
   outline: 1px solid var(--color-background-accent-two);
 }
 
-.note-card-title-and-pin {
+.note-card-priority-and-pin {
```
Comment (t007rushi) [inline src/frontend/components/NoteCard/NoteCard.css:14]: clear and understandable naming

File: src/frontend/components/NoteCard/NoteCard.js [hunk 2]
```diff
@@ -125,13 +136,15 @@ export const NoteCard = ({ note }) => {
     <div
       className={`${note.color} note-card`}
       onClick={() => {
-        if (pathname === "/notes") {
+        if (pathname === "/notes" || matchPath("/tags/*", pathname)) {
```
Comment (t007rushi) [inline src/frontend/components/NoteCard/NoteCard.js:139]: nice check

**Review by t007rushi — APPROVED** [review]
Yes check for empty label adding bug

**Pull request merged**